### ISSUE TITLE
this.Execute.EmbeddedScript fail with /* comment */ ?

### ISSUE THREAD

**Issue by jzabroski** [fluentmigrator/fluentmigrator#945]
I am not sure why this happens.

FM seems to like -- comments but dislike /* */ comments

**Comment by fubar-coder** [comment]
Which comments are allowed depend on the used database engine. Which DB engine do you use?

**Comment by jzabroski** [comment]
@fubar-coder Microsoft SQL Server

**Comment by jzabroski** [comment]
@fubar-coder In debugging this, the issue appears to be that I put example code inside the multi-line comments, and the example code contains a GO.
```sql
/*
-- Example usage:
EXEC dbo.sp_whoisactive
GO
*/
```

**Comment by jzabroski** [comment]
@fubar-coder For SQL Server, Microsoft has created a NuGet package for parsing the actual TSQL source tree. We could use this to improve how we split on 'GO' so that GO statements inside comments are ignored.

[Microsoft.Data.Tools.Sql.BatchParser](https://www.nuget.org/packages/Microsoft.Data.Tools.Sql.BatchParser/150.18040.0-xplat)

> The SQL Server Management Objects (SMO) Framework is a set of objects designed for programmatic management of Microsoft SQL Server and Microsoft Azure SQL Database.

[Microsoft.SqlServer.SqlParser](https://www.nuget.org/packages/Microsoft.SqlServer.SqlParser/150.18040.0-xplat)

> The SQL Server Management Objects (SMO) Framework is a set of objects designed for programmatic management of Microsoft SQL Server and Microsoft Azure SQL Database.

These are not yet open sourced on GitHub, but Microsoft seems to be moving in that direction.

**Comment by fubar-coder** [comment]
Yeah, that might work. I'll have to think about it (and the legal consequences).

**Comment by jzabroski** [comment]
I read through the license.  I can see why some users might not want it in the base install.  Could we make the SQL Parser a plug-in and have a separate DLL to include it?
